### PR TITLE
Improve `snow/` tests with `require`

### PR DESCRIPTION
**Pull request by dhrubabasu**
## Why this should be merged

Factored out of #1453 

## How this works

update tests

## How this was tested

CI

### CODE REVIEW

File: snow/engine/avalanche/vertex/test_builder.go [hunk 1]
```diff
@@ -33,7 +35,8 @@ func (b *TestBuilder) BuildStopVtx(ctx context.Context, parentIDs []ids.ID) (ava
 		return b.BuildStopVtxF(ctx, parentIDs)
 	}
 	if b.CantBuildVtx && b.T != nil {
-		b.T.Fatal(errBuild)
+		require := require.New(b.T)
+		require.FailNow(errBuild.Error())
```
Comment (danlaine) [inline snow/engine/avalanche/vertex/test_builder.go:39]:
```suggestion
		require.FailNow(b.T, errBuild.Error())
```

File: snow/engine/avalanche/vertex/test_parser.go [hunk 2]
```diff
@@ -32,7 +34,8 @@ func (p *TestParser) ParseVtx(ctx context.Context, b []byte) (avalanche.Vertex,
 		return p.ParseVtxF(ctx, b)
 	}
 	if p.CantParseVtx && p.T != nil {
-		p.T.Fatal(errParse)
+		require := require.New(p.T)
+		require.FailNow(errParse.Error())
```
Comment (danlaine) [inline snow/engine/avalanche/vertex/test_parser.go:38]:
```suggestion
		require.FailNow(p.T, errParse.Error())
```

Comment (danlaine) [reply snow/engine/avalanche/vertex/test_parser.go:38]: Similar comment for `test_storage.go` and other files below

**Review by danlaine — APPROVED** [review]
There's a bunch of places we do:

```
require := require.New(t)
require.DoSomething(...)
```
and that's the only time we use that `require`...I feel like we could just do `require.DoSomething(t, predicate)`. Idc that much though

**Review by joshua-kim — APPROVED** [review]

File: snow/consensus/snowball/binary_snowball_test.go [hunk 3]
```diff
@@ -109,53 +81,39 @@ func TestBinarySnowballAcceptWeirdColor(t *testing.T) {
...
-	if pref := sb.Preference(); pref != red {
-		t.Fatalf("Wrong preference. Expected %d got %d", red, pref)
-	} else if sb.Finalized() {
-		t.Fatalf("Finalized too early")
-	}
+	require.Equal(red, sb.Preference())
+	require.False(sb.Finalized())
 
 	sb.RecordSuccessfulPoll(red)
+
```
Comment (abi87) [inline snow/consensus/snowball/binary_snowball_test.go:94]: nit: remove to reduce diffs

File: snow/uptime/locked_calculator_test.go [hunk 4]
```diff
@@ -22,7 +22,7 @@ func TestLockedCalculator(t *testing.T) {
 	defer ctrl.Finish()
 
 	lc := NewLockedCalculator()
-	require.NotNil(t)
+	require.NotNil(lc)
```
Comment (abi87) [inline snow/uptime/locked_calculator_test.go:25]: nice catch

File: snow/engine/common/test_engine.go [hunk 5]
```diff
@@ -688,8 +693,11 @@ func (e *EngineTest) GetVM() VM {
 	if e.GetVMF != nil {
 		return e.GetVMF()
 	}
-	if e.CantGetVM && e.T != nil {
-		e.T.Fatalf("Unexpectedly called GetVM")
+	if !e.CantGetVM {
+		return nil
+	}
+	if e.T != nil {
+		require.FailNow(e.T, "Unexpectedly called GetVM")
```
Comment (abi87) [inline snow/engine/common/test_engine.go:700]: I believe this is not equivalent to originasl test

File: snow/engine/common/test_engine.go [hunk 6]
```diff
@@ -237,8 +239,11 @@ func (e *EngineTest) Halt(ctx context.Context) {
 		e.HaltF(ctx)
 		return
 	}
-	if e.CantHalt && e.T != nil {
-		e.T.Fatalf("Unexpectedly called Halt")
+	if !e.CantHalt {
+		return
+	}
+	if e.T != nil {
+		require.FailNow(e.T, "Unexpectedly called Halt")
```
Comment (abi87) [inline snow/engine/common/test_engine.go:246]: I believe this is not equivalent to the original test

File: snow/networking/tracker/resource_tracker_test.go [hunk 7]
```diff
@@ -66,28 +68,20 @@ func TestCPUTracker(t *testing.T) {
...
 
 	mockUser.EXPECT().CPUUsage().Return(.5).Times(3)
 
 	startTime3 := endTime2
 	endTime3 := startTime3.Add(halflife)
 	newNode1Utilization := cpuTracker.Usage(node1, endTime3)
-	if newNode1Utilization >= node1Utilization {
-		t.Fatalf("node CPU utilization should decrease over time")
-	}
+	require.Greater(node1Utilization, newNode1Utilization)
```
Comment (abi87) [inline snow/networking/tracker/resource_tracker_test.go:82]: q: what would be use for stricly greater? I just want to make sure we are not making the test stricter

File: snow/consensus/snowball/parameters_test.go [hunk 8]
```diff
@@ -74,9 +67,8 @@ func TestParametersInvalidK(t *testing.T) {
 		MaxItemProcessingTime: 1,
 	}
 
-	if err := p.Verify(); err == nil {
-		t.Fatalf("Should have failed due to invalid k")
-	}
+	err := p.Verify()
+	require.ErrorIs(t, err, ErrParametersInvalid)
```
Comment (abi87) [inline snow/consensus/snowball/parameters_test.go:71]: nice

File: snow/consensus/snowman/poll/early_term_no_traversal_test.go [hunk 9]
```diff
@@ -24,18 +28,13 @@ func TestEarlyTermNoTraversalResults(t *testing.T) {
...
 	result := poll.Result()
-	if list := result.List(); len(list) != 1 {
-		t.Fatalf("Wrong number of vertices returned")
-	} else if retVtxID := list[0]; retVtxID != vtxID {
-		t.Fatalf("Wrong vertex returned")
-	} else if result.Count(vtxID) != 1 {
-		t.Fatalf("Wrong number of votes returned")
-	}
+	list := result.List()
+	require.Len(list, 1)
```
Comment (abi87) [inline snow/consensus/snowman/poll/early_term_no_traversal_test.go:35]: we prolly don'y need the list variable

File: snow/consensus/snowman/poll/set_test.go [hunk 10]
```diff
@@ -234,36 +218,25 @@ func TestCreateAndFinishSuccessfulPoll(t *testing.T) {
...
-	} else if len(results) != 1 {
-		t.Fatalf("Wrong number of results returned")
-	} else if list := results[0].List(); len(list) != 1 {
-		t.Fatalf("Wrong number of vertices returned")
-	} else if retVtxID := list[0]; retVtxID != vtxID {
-		t.Fatalf("Wrong vertex returned")
-	} else if results[0].Count(vtxID) != 2 {
-		t.Fatalf("Wrong number of votes returned")
-	}
+	require.Zero(s.Len())
```
Comment (abi87) [inline snow/consensus/snowman/poll/set_test.go:221]: should this be require.Empty?

File: snow/engine/avalanche/bootstrap/bootstrapper_test.go [hunk 11]
```diff
@@ -997,7 +997,7 @@ func TestRestartBootstrapping(t *testing.T) {
 			}
 		default:
 			require.FailNow(errUnknownVertex.Error())
-			panic(errUnknownVertex)
+			return nil, errUnknownVertex
```
Comment (abi87) [inline snow/engine/avalanche/bootstrap/bootstrapper_test.go:1000]: nice

**Review by abi87 — CHANGES_REQUESTED** [review]
nits and questions

**Comment by abi87** [comment]
Kind of scary change overall. I wonder if, past the cleanup, we should run a mutation test campaign to make sure assertions are as strict as possible and we have not flipped some of the loose ones

**Review by coffeeavax — APPROVED** [review]

File: snow/networking/tracker/resource_tracker_test.go (hunk 7 above)
Comment (dhrubabasu) [reply snow/networking/tracker/resource_tracker_test.go:82]: This isn't making the test stricter, note the order of the variables compared to the original if statement.

File: snow/engine/common/test_engine.go (hunk 6 above)
Comment (dhrubabasu) [reply snow/engine/common/test_engine.go:246]: How so?

File: snow/consensus/snowman/poll/set_test.go (hunk 10 above)
Comment (dhrubabasu) [reply snow/consensus/snowman/poll/set_test.go:221]: It doesn't work for interfaces:

```
        	Error Trace:	/Users/dhruba/dev/avalanchego/snow/consensus/snowman/poll/set_test.go:221
        	Error:      	Should be empty, but was current polls: (Size = 0)
        	Test:       	TestCreateAndFinishSuccessfulPoll
```

File: snow/networking/benchlist/benchlist_test.go [hunk 12]
```diff
@@ -83,23 +76,23 @@ func TestBenchlistAdd(t *testing.T) {
 	}
 
 	// Still shouldn't be benched due to not enough consecutive failure
-	require.False(t, b.isBenched(vdrID0))
-	require.Zero(t, b.benchedQueue.Len())
-	require.Zero(t, b.benchlistSet.Len())
-	require.Len(t, b.failureStreaks, 1)
+	require.False(b.isBenched(vdrID0))
+	require.Zero(b.benchedQueue.Len())
+	require.Zero(b.benchlistSet.Len())
```
[the remark this answers is not in the document]
Comment (dhrubabasu) [reply snow/networking/benchlist/benchlist_test.go:81]: Yup, fixed.

File: snow/consensus/snowball/binary_snowball_test.go (hunk 3 above)
Comment (dhrubabasu) [reply snow/consensus/snowball/binary_snowball_test.go:94]: These gaps were done on purpose, I think it reads better

File: snow/engine/common/test_engine.go (hunk 6 above)
Comment (abi87) [reply snow/engine/common/test_engine.go:246]: I am wrong, this is good

**Review by abi87 — APPROVED** [review]

**Review by StephenButtolph — COMMENTED** [review]
all nits

**Review by StephenButtolph — APPROVED** [review]

**Pull request merged**